zrml/prediction-markets/src/migrations.rs

```diff
@@ -296,6 +447,52 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_cleanup_storage_for_resolved_or_closed_market_on_runtime_upgrade() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            let _ = AssetManager::deposit(Asset::Ztg, &ALICE, 1_000 * BASE);
+
+            let short_time: MomentOf<Runtime> = (5 * MILLISECS_PER_BLOCK).into();
+
+            create_test_market(
+                MarketPeriod::Timestamp(0..short_time),
+                MarketCreation::Permissionless,
+                MarketStatus::Resolved,
+            );
+
+            create_test_market(
+                MarketPeriod::Timestamp(0..short_time),
+                MarketCreation::Permissionless,
+                MarketStatus::Resolved,
+            );
+
+            System::set_block_number(1);
+            let market_ids = BoundedVec::<MarketIdOf<Runtime>, CacheSize>::try_from(vec![0, 1])
+                .expect("BoundedVec creation failed");
+            let dispute_block = System::current_block_number();
+            MarketIdsPerDisputeBlock::<Runtime>::insert(dispute_block, market_ids.clone());
+            MarketIdsPerReportBlock::<Runtime>::insert(dispute_block, market_ids.clone());
+            System::set_block_number(
+                System::current_block_number() + <Runtime as crate::Config>::DisputePeriod::get(),
+            );
+            assert_eq!(MarketIdsPerDisputeBlock::<Runtime>::get(dispute_block).len(), 2);
+            assert_eq!(MarketIdsPerReportBlock::<Runtime>::get(dispute_block).len(), 2);
+            CleanUpStorageForResolvedOrClosedMarkets::<Runtime>::on_runtime_upgrade();
+            assert_eq!(MarketIdsPerDisputeBlock::<Runtime>::get(dispute_block).len(), 0);
+            assert_eq!(MarketIdsPerReportBlock::<Runtime>::get(dispute_block).len(), 0);
+
+            let dispute_block = System::current_block_number();
+            MarketIdsPerDisputeBlock::<Runtime>::insert(dispute_block, market_ids.clone());
+            MarketIdsPerReportBlock::<Runtime>::insert(dispute_block, market_ids.clone());
+            System::set_block_number(System::current_block_number() + 1);
```

Comment:
I would probably go with the following.
```suggestion
            System::set_block_number(System::current_block_number() + <Runtime as crate::Config>::DisputePeriod::get() - 1);
```